lib/features/children/generosity_challenge_chat/chat_scripts/cubit/chat_scripts_cubit.dart

```diff
@@ -126,19 +153,21 @@ class ChatScriptsCubit extends Cubit<ChatScriptsState> {
           emit(state.copyWith(chatHistory: typingHead));
           await Future.delayed(_typingDuration, () {});
 
-          newHead = state.chatHistory.replaceHead(head: currentChatItem);
+          final formattedText = await _challengeCubit
+              .formatChatTextWithUserData(source: currentChatItem.text);
+
+          newHead = state.chatHistory
+              .replaceHead(head: currentChatItem.copyWith(text: formattedText));
         } else if (currentChatItem.isCondition &&
             state.gainedAnswer != const ChatScriptItem.empty()) {
-          final amplitudeEvent = state.gainedAnswer.amplitudeEvent;
-          if (amplitudeEvent.isNotEmpty) {
-            await AnalyticsHelper.logChatScriptEvent(
-              eventName: amplitudeEvent,
-              eventProperties: {
-                'value': state.gainedAnswer.answerText,
-              },
-            );
+          final saveKey =
+              ChatScriptSaveKey.fromString(state.gainedAnswer.saveKey);
+          if (saveKey.isSupported) {
+            await _challengeCubit.saveUserData(state.gainedAnswer);
```

Comment:
No checking for supported anymore, just an isEmpty check and then await _challengeCubit.saveUserData(state.gainedAnswer.saveKey);